CondTools/BeamSpot/plugins/BeamSpotOnlineFromOfflineConverter.cc

```diff
@@ -115,56 +117,61 @@ cond::Time_t BeamSpotOnlineFromOfflineConverter::pack(uint32_t fIOVStartRun, uin
 
 // ------------ method called for each event  ------------
 void BeamSpotOnlineFromOfflineConverter::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  const std::string fLabel = (fIsHLT_) ? "BeamSpotOnlineHLTObjectsRcd" : "BeamSpotOnlineLegacyObjectsRcd";
-  const BeamSpotObjects* inputSpot = &iSetup.getData(bsToken_);
-
-  BeamSpotOnlineObjects abeam;
-  abeam.copyFromBeamSpotObject(*inputSpot);
-  abeam.setLastAnalyzedLumi(lastAnalyzedLumi_);
-  abeam.setLastAnalyzedRun(lastAnalyzedRun_);
-  abeam.setLastAnalyzedFill(lastAnalyzedFill_);
-  abeam.setStartTimeStamp(std::time(nullptr));
-  abeam.setEndTimeStamp(std::time(nullptr));
-  abeam.setNumTracks(numTracks_);
-  abeam.setNumPVs(numPVs_);
-  abeam.setUsedEvents(numUsedEvents_);
-  abeam.setMaxPVs(numMaxPVs_);
-  abeam.setMeanPV(meanPVs_);
-  abeam.setMeanErrorPV(meanPVError_);
-  abeam.setRmsPV(rmsPV_);
-  abeam.setRmsErrorPV(rmsPVError_);
-  abeam.setStartTime(startTime_);
-  abeam.setEndTime(endTime_);
-  abeam.setLumiRange(lumiRange_);
-
-  // Set the creation time of the payload to the current time
-  auto creationTime =
-      std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
-  abeam.setCreationTime(creationTime);
-
-  edm::LogPrint("BeamSpotOnlineFromOfflineConverter") << " Writing results to DB...";
-  edm::LogPrint("BeamSpotOnlineFromOfflineConverter") << abeam;
-
-  edm::Service<cond::service::PoolDBOutputService> poolDbService;
-  if (poolDbService.isAvailable()) {
-    edm::LogPrint("BeamSpotOnlineFromOfflineConverter") << "poolDBService available";
-    if (poolDbService->isNewTagRequest(fLabel)) {
-      edm::LogPrint("BeamSpotOnlineFromOfflineConverter") << "new tag requested";
-      if (fuseNewSince_) {
-        edm::LogPrint("BeamSpotOnlineFromOfflineConverter") << "Using a new Since: " << fnewSince_;
-        poolDbService->createOneIOV<BeamSpotOnlineObjects>(abeam, fnewSince_, fLabel);
-      } else
-        poolDbService->createOneIOV<BeamSpotOnlineObjects>(abeam, poolDbService->beginOfTime(), fLabel);
-    } else {
-      edm::LogPrint("BeamSpotOnlineFromOfflineConverter") << "no new tag requested";
-      if (fuseNewSince_) {
-        edm::LogPrint("BeamSpotOnlineFromOfflineConverter") << "Using a new Since: " << fnewSince_;
-        poolDbService->appendOneIOV<BeamSpotOnlineObjects>(abeam, fnewSince_, fLabel);
-      } else
-        poolDbService->appendOneIOV<BeamSpotOnlineObjects>(abeam, poolDbService->currentTime(), fLabel);
+  if (bsWatcher_.check(iSetup)) {
+    const std::string fLabel = (fIsHLT_) ? "BeamSpotOnlineHLTObjectsRcd" : "BeamSpotOnlineLegacyObjectsRcd";
```

Comment:
I know this was already like that before, but this could go in the class constructor, right?

Reply:
sure will do in the next commit!

Reply:
addressed in last force-push commit: https://github.com/cms-sw/cmssw/pull/42909/commits/064118e709f10e968d446bb3d53662762a7e8606